Feat: Array result handler. (#1134)
RobinTail/express-zod-api: CHANGELOG.md

## Version 11

### v11.7.0

> - Good news for array lovers and those who struggles migrating their legacy APIs to using this library.

Author: ```suggestion
- Good news for array lovers and those struggling with migrating legacy APIs to use this library.
```

- New feature: `arrayResultHandler`.
  - Please avoid using it for new projects: responding with array is a bad practice keeping your endpoints from
    evolving without breaking changes.
  - This result handler expects your endpoint to have the property named `array` in its output schema.
  - The `array` property should be the `ZodArray` schema.

>   - The value of that property used as the response.

Author: ```suggestion
  - The value of that property is used as the response.
```

  - Missing the `array` property will result in internal error (status code `500`).
  - The negative response schema is `z.string()`, meaning that in case of error the response will be its plain message.

>   - The result handler also supports examples, being supported by documentation and client generators.

Author: ```suggestion
  - The result handler also supports examples, as well as documentation and client generation.
```
